resources/test/conftest.py

```diff
@@ -58,6 +69,27 @@ def repr_failure(self, excinfo):
         return pytest.Collector.repr_failure(self, excinfo)
 
     def runtest(self):
+        if self.type == 'unit':
+            self._run_unit_test()
+        elif self.type == 'functional':
+            self._run_functional_test()
+
+    def _run_unit_test(self):
+        driver = self.session.config.driver
+        server = self.session.config.server
+
+        driver.get(server.url(HARNESS))
+
+        actual = driver.execute_async_script('runTest("%s", "foo", arguments[0])' % server.url(str(self.filename)))
+
+        summarized = self._summarize(actual)
```

Comment:
Extract everything up to and including the `self._summarize()` call into another helper method? (Unless you have specific plans to diverge the two methods.)

Reply:
Yes, I do have plans for these two methods to diverge further